### PR TITLE
[GB-Mobile]: Add accessibilty label to the RichText component

### LINKED ISSUE #2364: When the page/post title is selected tell the user about that they are edditing the post title
**Describe the bug**
When a user is in VoiceOver mode and they create a new page to edit. They land on the Page Title field but it is not clear that they are editing the page title. 

**To Reproduce**
1. In the WP add enable VoiceOver, (using siri)
2. Create a new page. By selecting the + and double clicking. 
3. Notice that VoiceOver never metions that you are editing the Page title.
 
**Expected behavior**
When you land on the page 

**Screenshots**
Here is a quick video. 
https://cloudup.com/cNlMs6Fbu_F

**Smartphone (please complete the following information):**
 - Device: iPhone Pro 11
 - OS: iOS 13.5
 - Version: 15

**Additional context**

### PR DESCRIPTION
**Pull request by enejb**
Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/2364
Related Gutenberg Mobile PR:
https://github.com/wordpress-mobile/gutenberg-mobile/pull/2368

## Description
This makes it possible PR makes it possible to pass the accessibilityLabel and accessibilityHint to the RichText component. Which helps us set the correct label on the input field. 

## How has this been tested?
This was tested using VoiceOver on iOS devices. 
See instruction in https://github.com/wordpress-mobile/gutenberg-mobile/pull/2368 

## Screenshots 
https://cloudup.com/cjd-RHZpyQM

## Types of changes
This PR 
- updates the accessibility Label so that it is Less post type specifici. We just call the field Title instead of "Post title" when the user is editing a page.
- Update the accessibility Label to be more helpful when the Element is selected. 
 
## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by enejb** [comment]
Pinging @chipsnyder  and @pinarol for a review :)

**Comment by enejb** [comment]
I am not sure why the tests are failing. They don't seem to be related to the change.

File: packages/editor/src/components/post-title/index.native.js [hunk 1]
```diff
@@ -91,15 +91,24 @@ class PostTitle extends Component {
...
 			>
 				<RichText
+					accessibilityLabel={
+						isEmpty( title )
+							? /* translators: accessibility text. empty post title. */
+							  __( 'Title. Empty' )
+							: /* translators: accessibility text. Name of the form field. */
+							  __( 'Title.' )
+					}
+					accessibilityHint={ __( 'Update the title.' ) }
```
Comment (etoledom) [inline packages/editor/src/components/post-title/index.native.js:111]: From https://developer.apple.com/documentation/objectivec/nsobject/1615093-accessibilityhint
> Avoid beginning the phrase with the imperative form of a verb, because this can make the hint sound like a command

```suggestion
					accessibilityHint={ __( 'Updates the title.' ) }
```

Even though still is weird since executing the action won't automatically "update the title", but I guess the intention is understood? 🤔

File: packages/editor/src/components/post-title/index.native.js [hunk 2]
```diff
@@ -91,15 +91,24 @@ class PostTitle extends Component {
 				accessibilityLabel={
 					isEmpty( title )
 						? /* translators: accessibility text. empty post title. */
-						  __( 'Post title. Empty' )
+						  __( 'Title. Empty' )
 						: sprintf(
 								/* translators: accessibility text. %s: text content of the post title. */
-								__( 'Post title. %s' ),
+								__( 'Title. %s' ),
```
Comment (etoledom) [inline packages/editor/src/components/post-title/index.native.js:97]: Just wondering, what's the rationale for this change?

**Review by etoledom — COMMENTED** [review]
Thank you @enejb for improving accessibility! These PRs are always welcome 🎉 

I do have some thoughts about adding extra labels and hints to Aztec's text view, I explained them as code comments. I'd love to know your opinion on those.

One more thing:
It feels like it's missing the action. We generally set the accessibility role to `Button` to let the user that the element is tappable.

File: packages/rich-text/src/component/index.native.js [hunk 3]
```diff
@@ -765,6 +767,8 @@ export class RichText extends Component {
 						onFocus: () => {},
 					} ) }
 				<RCTAztecView
+					accessibilityLabel={ accessibilityLabel }
+					accessibilityHint={ accessibilityHint }
```
Comment (etoledom) [inline packages/rich-text/src/component/index.native.js:771]: Some thoughts about this here: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2368#discussion_r439370912

**Comment by etoledom** [comment]
> I am not sure why the tests are failing. They don't seem to be related to the change.

I have restarted the failing test and now it passed 🎉 
We get used to these things 😁

File: packages/editor/src/components/post-title/index.native.js (hunk 2 above)
Comment (enejb) [reply packages/editor/src/components/post-title/index.native.js:97]: Since they could be editing the "Page Title" I wanted to remove the `post_type` part. It would also make things shorter. 

Should we instead take the post_type into account and say "Page title" or "Post title" depending on the post_type that the user is editing?

File: packages/rich-text/src/component/index.native.js (hunk 3 above)
Comment (enejb) [reply packages/rich-text/src/component/index.native.js:771]: Should we remove the hint completely?

File: packages/editor/src/components/post-title/index.native.js (hunk 2 above)
Comment (etoledom) [reply packages/editor/src/components/post-title/index.native.js:97]: > Since they could be editing the "Page Title"

Good point!

> Should we instead take the post_type into account and say "Page title" or "Post title" depending on the post_type that the user is editing?

Maybe being specific is good, since "Title" might be confused with "Headings" by users.

File: packages/rich-text/src/component/index.native.js (hunk 3 above)
Comment (etoledom) [reply packages/rich-text/src/component/index.native.js]: I think so 🤔 
The hint doesn't seem to add much when the user is already editing the TextView. Let's keep the hint on the Wrapper view and remove it from Aztec.

Does it sound good?

Comment (enejb) [reply packages/rich-text/src/component/index.native.js]: Yeah, sounds like a good idea.

File: packages/editor/src/components/post-title/index.native.js (hunk 2 above)
Comment (enejb) [reply packages/editor/src/components/post-title/index.native.js:97]: Can you explain a bit more about what your concern when it comes to Headings? 

I noticed that in the web side the placeholder says "Add title" for both post and pages.

Comment (etoledom) [reply packages/editor/src/components/post-title/index.native.js:97]: Opening a new post, VoiceOver focuses on the Title, and speaks out that the TextField it's focused is a Title.
That's good and works for both post and pages.

My only concern is the user expectation or knowledge about what this title is. They don't have the visual clue for it to be the post wide title or a section/paragraph "title". It might be confused with a content Heading. So "Post Title" and "Page Title" would cue better that this is the global title.

It's not a strong opinion, just something we could consider. I might be overthinking it :)
Let me know if you prefer to leave as it is to ✅ and :shipit: 👍

Comment (enejb) [reply packages/editor/src/components/post-title/index.native.js:97]: My preferences is to just use "Title" it as it is. But I don't have strong feeling about this. 
But I do agree that it makes generic. Will try to update it to use "Post title" and "Page title"

**Review by etoledom — APPROVED** [review]
Approving since it's working as expected 

> Will try to update it to use "Post title" and "Page title"

@enejb not sure what happened with this, let me know if you are going to update to give it a final look 👀

**Comment by enejb** [comment]
Sorry @etoledom 
I switched focuses for a bit will try to get this updated tomorrow.

**Review by etoledom — APPROVED** [review]
It's working great 🎉 

Thank you for the update!
